modules/vidloop/vidloop.c

```diff
@@ -164,16 +188,18 @@ static int display(struct video_loop *vl, struct vidframe *frame)
 	vl->disp_size = frame->size;
 	++vl->stats.disp_frames;
 
-	/* display frame */
-	err = vidisp_display(vl->vidisp, "Video Loop", frame);
-	if (err == ENODEV) {
-		info("vidloop: video-display was closed\n");
-		vl->vidisp = mem_deref(vl->vidisp);
-		vl->err = err;
-		goto out;
+	pthread_mutex_lock(&vl->frame_mutex);
+
```

Comment:
check for changes in frame size here. if the size changed, you can just `mem_deref` the
current frame.